File: bindings/js/src/core/domain/shapes/PropertyShape.ts
/**
 * Created by antoniogarrote on 10/05/2017.
 */
import {SH_NS, Shape, SHAPES_NS, XSDDataType} from "./Shape";

export class PropertyShape extends Shape {

    public getPath(): string | undefined {
        return this.getRdfId(`${SH_NS}path`);
    }

    public setPath(path: string | null) {
        this.setRdfId(`${SH_NS}path`, path);
    }

    public getPropertyLabel(): string | undefined {
        return this.getValue<string>(`${SHAPES_NS}propertyLabel`);
    }

    public setPropertyLabel(label: string | null) {
        this.setValue(`${SHAPES_NS}propertyLabel`, label);
    }

    public getMinCount(): number | undefined {
        return this.getValue<number>(`${SH_NS}minCount`);
    }

    public setMinCount(count: number | null) {
        this.setValue(`${SH_NS}minCount`, count);
    }

    public getMaxCount(): number | undefined {
        return this.getValue<number>(`${SH_NS}maxCount`);
    }

    public setMaxCount(count: number | null) {
        this.setValue(`${SH_NS}maxCount`, count);
    }


    public getDatatype(): XSDDataType | undefined {
        return this.getRdfId(`${SH_NS}datatype`);
    }

    public setDatatype(type: XSDDataType | null) {
        this.setRdfId(`${SH_NS}datatype`, type);
    }

    public getNode(): Shape | undefined {
        const object = this.getRdfObject(`${SH_NS}node`);
        if (object != null)
            return Shape.fromRawModel(object);
    }

    public setNode(object: Shape | null) {
        if (object != null) {
            this.setRdfObject(`${SH_NS}node`, object.clojureModel());
        } else {
            this.setRdfObject(`${SH_NS}node`, null);
        }
    }

    public getOrdered(): boolean {
        return this.getValue<boolean>(`${SHAPES_NS}ordered`) || false;
    }

    public setOrdered(ordered: boolean | null) {
        this.setValue(`${SH_NS}ordered`, ordered);
    }

    public static build(id: string): PropertyShape {
        return new PropertyShape({"@id": id, "@type": [`${SH_NS}Shape`, `${SH_NS}PropertyShape`]})
    }
}

File: bindings/java/src/org/raml/amf/core/domain/shapes/PropertyShape.java
package org.raml.amf.core.domain.shapes;

import org.raml.amf.utils.Clojure;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by antoniogarrote on 15/05/2017.
 */

public class PropertyShape extends Shape {

    public PropertyShape(Object rawModel) {
        super(rawModel);
    }

    public String getPath() {
        return this.getRdfId(SH_NS + "path");
    }

    public void setPath(String path) {
        this.setRdfId(SH_NS + "path", path);
    }

    public String getPropertyLabel() {
        return this.getValue(SHAPES_NS + "propertyLabel");
    }

    public void setPropertyLabel(String label) {
        this.setValue(SHAPES_NS + "propertyLabel", label);
    }

    public Integer getMinCount() {
        return this.getValue(SH_NS + "minCount");
    }

    public void setMinCount(Integer count) {
        this.setValue(SH_NS + "minCount", count);
    }

    public Integer getMaxCount() {
        return this.getValue(SH_NS + "maxCount");
    }

    public void setMaxCount(Integer count) {
        this.setValue(SH_NS + "maxCount", count);
    }

    public String getDatatype() {
        return this.getRdfId(SH_NS + "datatype");
    }

    public void setDatatype(String type) {
        this.setRdfId(SH_NS + "datatype", type);
    }

    public Shape getNode() {
        Object object = this.getRdfObject(SH_NS + "node");
        if (object != null)
            return Shape.fromRawModel(object);
        return null;
    }

    public void setNode(Shape object) {
        if (object != null) {
            this.setRdfObject(SH_NS + "node", object.clojureModel());
        } else {
            this.setRdfObject(SH_NS + "node", null);
        }
    }

    public boolean getOrdered() {
        Boolean res = this.getValue(SHAPES_NS + "ordered");
        if (res == null) {
            return false;
        } else {
            return res;
        }
    }

    public void setOrdered(Boolean ordered) {
        this.setValue(SH_NS + "ordered", ordered);
    }

    public static PropertyShape build(String id) {
        Object acc = Clojure.emptyMap();
        acc = Clojure.set(acc, "@id", id);
        List<String> types = new ArrayList<>();
        types.add(id);
        acc = Clojure.set(acc, "@type", Clojure.list(types));
        return new PropertyShape(acc);
    }
}
